src/Core/Domain/Product/ValueObject/ProductCondition.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+
+class ProductCondition
+{
+    const NEW = 'new';
+    const USED = 'used';
+    const REFURBISHED = 'refurbished';
+
+    /**
+     * @var string
+     */
+    private $value;
+
+    /**
+     * @param string $value
+     */
+    public function __construct(string $value)
+    {
+        $this->assertValueIsAllowed($value);
+        $this->value = $value;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    /**
+     * @param string $value
+     */
+    private function assertValueIsAllowed(string $value): void
+    {
+        $allowedConditions = [
+            self::NEW,
+            self::USED,
+            self::REFURBISHED,
+        ];
+
+        if (!in_array($value, $allowedConditions, true)) {
+            throw new ProductConstraintException(
+                sprintf(
+                    'Invalid product condition "%s". Allowed conditions are: "%s"',
+                    $value,
+                    implode(',', $allowedConditions)
+                ),
+            ProductConstraintException::INVALID_CONDITION
```

Comment:
```suggestion
                ProductConstraintException::INVALID_CONDITION
```
No? It's weird CS fixer doesn't bug on this ^^